jax/core.py

```diff
@@ -384,19 +384,22 @@ def full_raise(self, val) -> 'Tracer':
       elif val._trace.sublevel < sublevel:
         return self.sublift(val)
       else:
-        raise escaped_tracer_error("Can't lift sublevels {} to {}"
-                                   .format(val._trace.sublevel, sublevel))
+        raise escaped_tracer_error(
+            f"Can't lift sublevels {val._trace.sublevel} to {sublevel}."
+            + tracer_line_info_detail(val))
     elif val._trace.level < level:
       if val._trace.sublevel > sublevel:
-        raise escaped_tracer_error("Incompatible sublevel: {}, {}"
-                                   .format(val._trace, (level, sublevel)))
+        raise escaped_tracer_error(
+            f"Incompatible sublevel: {val._trace}, {(level, sublevel)}."
+            + tracer_line_info_detail(val))
       return self.lift(val)
     elif val._trace.level > level:
-      raise escaped_tracer_error("Can't lift level {} to {}"
-                                 .format(val, self))
+      raise escaped_tracer_error(f"Can't lift level {val} to {self}."
+                                 + tracer_line_info_detail(val))
     else:  # val._trace.level == self.level:
-      raise escaped_tracer_error("Different traces at same level: {}, {}"
-                                 .format(val, self))
+      raise escaped_tracer_error(
+          f"Different traces at same level: {val}, {self}."
+          + tracer_line_info_detail(val))
```

Comment:
Factor the common call to `tracer_line_info_detail` into `escaped_tracer_error`? The latter is a helper function already. Up to you.

Reply:
It's a good idea, though `escaped_tracer_error` is called in a couple other places. Those error messages are somewhat related, so I'll take a look and see if updating them makes sense.